sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -639,12 +649,21 @@ def add_interval(
             "dev interval" if is_dev else "interval",
             start_ts,
             end_ts,
-            snapshot.snapshot_id,
+            snapshot_id,
+        )
+        snapshot_intervals = SnapshotIntervals(
```

Comment:
Should we just make the new method accept `SnapshotIntervals` with populated `intervals` / `dev_intervals`? This will make the API cleaner IMHO

Reply:
eg:
```
def add_snapshot_intervals(self, intervals: SnapshotIntervals)
```

Reply:
ya i like that

Reply:
https://github.com/TobikoData/sqlmesh/pull/2543/commits/016a36b60d9beaa14107e767fcf31d715e22ef3f